### PR TITLE
fix: folder assignment

### LINKED ISSUE #10448: Folders are not assigned after account creation
### Checklist

- [x] I have used the search function to see if someone else has already submitted the same bug report.
- [x] I will describe the problem with as much detail as possible.

### App

Thunderbird for Android

### App version

Daily 17.x

### Where did you get the app from?

Other

### Android version

Not relevant

### Device model

Not releavant

### Steps to reproduce

1. Go to "Setting"
2. Add a new account
3. Finish the account setup

### Expected behavior

When opening the drawer all special folders (inbox, outbox, drafts, sent, spam and trash) show their corresponding icon.

### Actual behavior

Occasionally, special folders appear as standard folders with a default icon. This visual effect reflects an underlying data issue that prevents newly added accounts from working correctly, particularly when the inbox isn't configured properly. The settings menu doesn't offer a way to fix a missing inbox, effectively breaking the account.

### Logs

_No response_

### PR DESCRIPTION
**Pull request by wmontwe**
Resolves #10448

This pull request introduces a synchronous update mechanism for special mail folders and refactors the `DefaultSpecialFolderUpdater` to support both asynchronous and synchronous updates. It also adds unit tests to ensure correct folder updating behavior, especially for edge cases like POP3 accounts and imported folders.

- Added a new `updateSpecialFoldersSync()` method to the `SpecialFolderUpdater` interface and its implementations, allowing for synchronous updating of special folders. This is particularly important for scenarios where immediate consistency is required, such as after an account creation.
- Refactored `DefaultSpecialFolderUpdater` to separate synchronous (`updateSpecialFoldersSync`) and asynchronous (`updateSpecialFolders`) update logic. The synchronous logic uses `runBlocking` to ensure updates complete before proceeding, and the asynchronous logic delegates to the synchronous method within a coroutine.
- Improved the logic for updating special folders: now, if a folder is set to manual selection but no longer exists, it falls back to automatic selection.

For better testability the `FolderRepository` is split into an interface and a default implementation.

### CODE REVIEW

File: legacy/mailstore/src/main/java/app/k9mail/legacy/mailstore/FolderRepository.kt [hunk 1]
```diff
@@ -1,218 +1,149 @@
...
-                isIntegrate = folder.isIntegrate,
-                isSyncEnabled = folder.isSyncEnabled,
-                isVisible = folder.isVisible,
-                isNotificationsEnabled = folder.isNotificationsEnabled,
-                isPushEnabled = folder.isPushEnabled,
-            )
-        }
-    }
-
+interface FolderRepository {
```
Comment (rafaeltonholo) [inline legacy/mailstore/src/main/java/app/k9mail/legacy/mailstore/FolderRepository.kt:18]: suggestion: Could you please add a TODO comment to make all methods that perform IO operations into suspend functions.

File: legacy/core/src/main/java/com/fsck/k9/mailstore/DefaultSpecialFolderUpdater.kt [hunk 2]
```diff
@@ -21,33 +22,58 @@ import net.thunderbird.feature.mail.folder.api.SpecialFolderUpdater
...
     private val accountId: AccountId,
     private val coroutineScope: CoroutineScope,
     private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO,
 ) : SpecialFolderUpdater {
-    override fun updateSpecialFolders() {
-        coroutineScope.launch(ioDispatcher) {
-            var account: LegacyAccount = getAccountById(accountId)
-            val folders = folderRepository.getRemoteFolders(accountId)
+    private fun getSpecialFolderSelection(account: LegacyAccount, type: FolderType) = when (type) {
+        FolderType.ARCHIVE -> account.archiveFolderSelection
```
Comment (rafaeltonholo) [inline legacy/core/src/main/java/com/fsck/k9/mailstore/DefaultSpecialFolderUpdater.kt]: The `.copy()` isn't needed anymore?

Comment (wmontwe) [reply legacy/core/src/main/java/com/fsck/k9/mailstore/DefaultSpecialFolderUpdater.kt]: This is not needed.

File: legacy/mailstore/src/main/java/app/k9mail/legacy/mailstore/FolderRepository.kt (hunk 1 above)
Comment (wmontwe) [reply legacy/mailstore/src/main/java/app/k9mail/legacy/mailstore/FolderRepository.kt:18]: I added a general TODO as the `FolderRepository` mixes responsibilities.
